array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,29 @@
+/* exported zip */
+function zip(first, second) {
+  const mainArr = [];
+  if (first.length > second.length) {
+    while (first.length !== second.length) {
+      first.pop();
+    }
+  }
```

Comment:
Instead of changing the inputs, figure out which array is shorter with Math.min

```js
const length = Math.min(first.length, second.length);
```

then loop until `length` and push first at `i` and second at `i` into the output

```js
for (let i = 0; i < length; i++) {
  // get first[i] and second[i] and push them into the array as a pair
}
```